File: Java/src/classi/Studente.java
package classi;

import java.time.LocalDate;

public class Studente {
    // attributi della classe. Per seguire la regola dell'incapsulamento, mettiamo tutti gli attributi private
    private String nome;
    private int anno;
    private String matricola;
    private String classe;

    // metodo costruttore -> permette di creare (istanziare) l'oggetto
    // di norma public. non ha valore di ritorno. Deve avere lo stesso nome della classe
    public Studente(String nome, int anno, String matricola, String classe) {
        this.nome = nome;
        this.anno = anno;
        this.matricola = matricola;
        this.classe = classe;
    }

    public int calcolo_eta() {
        return LocalDate.now().getYear() - anno;
    }

    public void stampa() {
        System.out.println("nome: " + nome + ", Anno: " + anno + ", Matricola: " + matricola + ", classe: " + classe);
    }

    // getter
    public String getNome() {
        return nome;
    }
    public int getAnno() {
        return anno;
    }
    public String getMatricola() {
        return matricola;
    }
    public String getClasse() {
        return classe;
    }

    // setter
    // per gli altri attributi non metto il setter perchè non ha senso logico poter modificare quei valori di uno studente 
    public void setClasse(String classe) {
        this.classe = classe;
    }
}

File: python/esempi_corso_start2data/classi/studente.py
from datetime import datetime


class Studente:
    def __init__(self, nome: str, anno: int, matricola: str, maschio: bool = True):
        self.nome = nome
        self.anno = anno
        self.matricola = matricola
        self.maschio = maschio

    def calcola_eta(self) -> int:
        return datetime.now().year - self.anno

    def __str__(self) -> str:
        return f"nome: {self.nome}, anno: {self.anno}, matricola: {self.matricola}"

    # stiamo definendo la somma: non ha assulatamente senso logico, è solo per mostrare il meme
    def __add__(self, studente: "Studente") -> "Studente":
        return Studente(
            self.nome + studente.nome, (self.anno + studente.anno) // 2, self.matricola + studente.matricola
        )
